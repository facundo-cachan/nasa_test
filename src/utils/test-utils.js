import React from 'react'
import { render } from '@testing-library/react-native';
import { AppProvider } from '@navigation/AppProvider';

const AllTheProviders = ({ children }) => {
    return <AppProvider>{children}</AppProvider>;
};

const customRender = (ui, options) =>
    render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
