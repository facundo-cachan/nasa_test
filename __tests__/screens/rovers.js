import React from 'react';
import userEvent from '@testing-library/user-event';
import { fireEvent, cleanup, render, waitFor, act } from 'test-utils';

import RoversScreen from '@screens/rovers';
jest.useFakeTimers();
const navigation = {
    navigate: jest.fn(),
},
    params = { id: 'spirit', camera: 'fhaz' };

describe('Testing Screens', () => {
    afterEach(cleanup);
    test('Rovers test', async () => {
        jest.runOnlyPendingTimers();
        const { baseElement, getByTestId } = render(
            <RoversScreen navigation={navigation} route={params} />,
        );
        // expect(getByTestId('RoversScreen')).toBeTruthy();
        expect(baseElement).toMatchSnapshot();
    });
});
