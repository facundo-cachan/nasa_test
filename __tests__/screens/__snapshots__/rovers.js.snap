// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screens Rovers test 1`] = `undefined`;
